src/runtime/importhook.cs

```diff
@@ -70,25 +76,25 @@ internal static void Initialize()
 
         internal static void Shutdown()
         {
-#if PYTHON3
             if (0 != Runtime.Py_IsInitialized()) {
+#if PYTHON3
                 Runtime.XDecref(py_clr_module);
-                Runtime.XDecref(root.pyHandle);
-            }
-            ModuleDefOffset.FreeModuleDef(module_def);
 #elif PYTHON2
-            if (0 != Runtime.Py_IsInitialized())
-            {
-                Runtime.XDecref(root.pyHandle);
                 Runtime.XDecref(root.pyHandle);
-            }
 #endif
-            if (0 != Runtime.Py_IsInitialized())
-            {
+                Runtime.XDecref(root.pyHandle);
                 Runtime.XDecref(py_import);
             }
         }
 
+        internal static void Cleanup()
```

Comment:
Cool, after it's removed I'm ok merging it. This fixes the issue for Python2.7, but it still occassionally fails on Python3 but it seems to be due to a different section of code